snorql-framework/src/main/kotlin/com/udaan/snorql/framework/job/JobManager.kt

```diff
@@ -0,0 +1,285 @@
+                    description,
+                    configuredByName,
+                    configuredByEmail
+                )
+            configSuccess
+        } catch (e: Exception) {
+            print("Unable to add data recording: $e")
+            false
+        }
+    }
+
+    /**
+     * Fetch historical data for a metricId and databaseName
+     */
+    fun getHistoricalData(
+        metricId: String,
+        databaseName: String,
+        pageNumber: Int,
+        pageSize: Int,
+        params: Map<String, *> = mapOf<String, String>()
+    ): List<HistoricalDatabaseSchemaDTO> {
+        return SqlMetricManager.queryExecutor.fetchHistoricalData(metricId, databaseName, pageNumber, pageSize, params)
+    }
+
+    private fun <T : MetricInput, O : IMetricResult, V : IMetricRecommendation> configureJobAndTrigger(
+        metricInput: T,
+        intervalInSeconds: Int,
+        endAt: Timestamp?,
+        description: String?,
+        configuredByName: String?,
+        configuredByEmail: String?
+    ): Boolean {
+        val jobName: String = metricInput.metricId // jobName = metricId (Therefore, for each metric, there is a job
+        val triggerName: String =
+            metricInput.metricId.plus("_").plus(metricInput.databaseName).plus("_")
+                .plus(SnorqlConstants.objectMapper.writeValueAsString(metricInput).hashCode())
+        val jobDataMap = JobDataMap()
+        jobDataMap["metricInput"] =
+            objectMapper.writeValueAsString(metricInput) // gson.toJson(metricInput).toString() // Use Jackson
+        jobDataMap["inputClass"] = metricInput::class.java.name
+        jobDataMap["configuredByName"] = configuredByName
+        jobDataMap["configuredByEmail"] = configuredByEmail
+        jobDataMap["repeatInterval"] = intervalInSeconds
+        val jobKey = JobKey(jobName, SnorqlConstants.DATA_PERSISTENCE_GROUP_NAME)
+        val triggerKey = TriggerKey(triggerName, SnorqlConstants.DATA_PERSISTENCE_GROUP_NAME)
+        return if (!scheduler.checkExists(jobKey)) {
+            println("Job does not exist. Configuring a job with job key $jobKey")
+            val job = JobBuilder.newJob(DataPersistenceJob<T, O, V>().javaClass)
+                .withIdentity(jobName, SnorqlConstants.DATA_PERSISTENCE_GROUP_NAME)
+                .storeDurably()
+                .build()
+            val trigger: SimpleTrigger =
+                TriggerBuilder.newTrigger()
+                    .withIdentity(triggerName, SnorqlConstants.DATA_PERSISTENCE_GROUP_NAME)
+                    .withDescription(description)
+//                    .startAt(startFrom)
+                    .usingJobData(jobDataMap)
+                    .withSchedule(
+                        SimpleScheduleBuilder.simpleSchedule()
+                            .withIntervalInSeconds(intervalInSeconds)
+                            .repeatForever()
+                    )
+                    .endAt(endAt)
+                    .build()
```

Comment:
This code seems to be repeating. Can this be extracted to a separate method?

Reply:
Resolved.
Added  function buildTriggerObject in JobManager to generate a Simple Trigger Object